File: sdks/java/lib/src/main/java/models/shared/File.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * File
 * The bare basic representation of a file, the minimal
 * amount of fields returned when using the `fields` query
 * parameter.
**/
public class File {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("content_created_at")
    public OffsetDateTime contentCreatedAt;
    public File withContentCreatedAt(OffsetDateTime contentCreatedAt) {
        this.contentCreatedAt = contentCreatedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("content_modified_at")
    public OffsetDateTime contentModifiedAt;
    public File withContentModifiedAt(OffsetDateTime contentModifiedAt) {
        this.contentModifiedAt = contentModifiedAt;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public File withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserMini createdBy;
    public File withCreatedBy(UserMini createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonProperty("description")
    public String description;
    public File withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("etag")
    public String etag;
    public File withEtag(String etag) {
        this.etag = etag;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("file_version")
    public FileVersionMini fileVersion;
    public File withFileVersion(FileVersionMini fileVersion) {
        this.fileVersion = fileVersion;
        return this;
    }
    @JsonProperty("id")
    public String id;
    public File withId(String id) {
        this.id = id;
        return this;
    }
    @JsonProperty("item_status")
    public FileItemStatusEnum itemStatus;
    public File withItemStatus(FileItemStatusEnum itemStatus) {
        this.itemStatus = itemStatus;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("modified_at")
    public OffsetDateTime modifiedAt;
    public File withModifiedAt(OffsetDateTime modifiedAt) {
        this.modifiedAt = modifiedAt;
        return this;
    }
    @JsonProperty("modified_by")
    public FileUserBase modifiedBy;
    public File withModifiedBy(FileUserBase modifiedBy) {
        this.modifiedBy = modifiedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public File withName(String name) {
        this.name = name;
        return this;
    }
    @JsonProperty("owned_by")
    public FileUserBase ownedBy;
    public File withOwnedBy(FileUserBase ownedBy) {
        this.ownedBy = ownedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("parent")
    public FileFolderBase parent;
    public File withParent(FileFolderBase parent) {
        this.parent = parent;
        return this;
    }
    @JsonProperty("path_collection")
    public FilePathCollection pathCollection;
    public File withPathCollection(FilePathCollection pathCollection) {
        this.pathCollection = pathCollection;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("purged_at")
    public OffsetDateTime purgedAt;
    public File withPurgedAt(OffsetDateTime purgedAt) {
        this.purgedAt = purgedAt;
        return this;
    }
    @JsonProperty("sequence_id")
    public java.util.Map<String, Object> sequenceId;
    public File withSequenceId(java.util.Map<String, Object> sequenceId) {
        this.sequenceId = sequenceId;
        return this;
    }
    @JsonProperty("sha1")
    public String sha1;
    public File withSha1(String sha1) {
        this.sha1 = sha1;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("shared_link")
    public FileSharedLink sharedLink;
    public File withSharedLink(FileSharedLink sharedLink) {
        this.sharedLink = sharedLink;
        return this;
    }
    @JsonProperty("size")
    public Long size;
    public File withSize(Long size) {
        this.size = size;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("trashed_at")
    public OffsetDateTime trashedAt;
    public File withTrashedAt(OffsetDateTime trashedAt) {
        this.trashedAt = trashedAt;
        return this;
    }
    @JsonProperty("type")
    public FileTypeEnum type;
    public File withType(FileTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/file.go
package shared

import (
	"time"
)

type FileItemStatusEnum string

const (
	FileItemStatusEnumActive  FileItemStatusEnum = "active"
	FileItemStatusEnumTrashed FileItemStatusEnum = "trashed"
	FileItemStatusEnumDeleted FileItemStatusEnum = "deleted"
)

type FileUserBaseTypeEnum string

const (
	FileUserBaseTypeEnumUser FileUserBaseTypeEnum = "user"
)

// FileUserBase
// The user who last modified this file
type FileUserBase struct {
	ID    *string              `json:"id,omitempty"`
	Login string               `json:"login"`
	Name  string               `json:"name"`
	Type  FileUserBaseTypeEnum `json:"type"`
}

type FileFolderBaseTypeEnum string

const (
	FileFolderBaseTypeEnumFolder FileFolderBaseTypeEnum = "folder"
)

// FileFolderBase
// The folder that this file is located within.
type FileFolderBase struct {
	Etag       *string                `json:"etag,omitempty"`
	ID         string                 `json:"id"`
	Name       *string                `json:"name,omitempty"`
	SequenceID map[string]interface{} `json:"sequence_id,omitempty"`
	Type       FileFolderBaseTypeEnum `json:"type"`
}

// FilePathCollection
// The tree of folders that this file is contained in,
// starting at the root.
type FilePathCollection struct {
	Entries    []FolderMini `json:"entries"`
	TotalCount int64        `json:"total_count"`
}

type FileSharedLinkAccessEnum string

const (
	FileSharedLinkAccessEnumOpen          FileSharedLinkAccessEnum = "open"
	FileSharedLinkAccessEnumCompany       FileSharedLinkAccessEnum = "company"
	FileSharedLinkAccessEnumCollaborators FileSharedLinkAccessEnum = "collaborators"
)

type FileSharedLinkEffectiveAccessEnum string

const (
	FileSharedLinkEffectiveAccessEnumOpen          FileSharedLinkEffectiveAccessEnum = "open"
	FileSharedLinkEffectiveAccessEnumCompany       FileSharedLinkEffectiveAccessEnum = "company"
	FileSharedLinkEffectiveAccessEnumCollaborators FileSharedLinkEffectiveAccessEnum = "collaborators"
)

type FileSharedLinkEffectivePermissionEnum string

const (
	FileSharedLinkEffectivePermissionEnumCanEdit     FileSharedLinkEffectivePermissionEnum = "can_edit"
	FileSharedLinkEffectivePermissionEnumCanDownload FileSharedLinkEffectivePermissionEnum = "can_download"
	FileSharedLinkEffectivePermissionEnumCanPreview  FileSharedLinkEffectivePermissionEnum = "can_preview"
	FileSharedLinkEffectivePermissionEnumNoAccess    FileSharedLinkEffectivePermissionEnum = "no_access"
)

// FileSharedLinkPermissions
// Defines if this link allows a user to preview, edit, and download an item.
// These permissions refer to the shared link only and
// do not supersede permissions applied to the item itself.
type FileSharedLinkPermissions struct {
	CanDownload bool `json:"can_download"`
	CanEdit     bool `json:"can_edit"`
	CanPreview  bool `json:"can_preview"`
}

// FileSharedLink
// The shared link for this file. This will be
// `null` if no shared link has been created for this
// file.
type FileSharedLink struct {
	Access              *FileSharedLinkAccessEnum             `json:"access,omitempty"`
	DownloadCount       int64                                 `json:"download_count"`
	DownloadURL         *string                               `json:"download_url,omitempty"`
	EffectiveAccess     FileSharedLinkEffectiveAccessEnum     `json:"effective_access"`
	EffectivePermission FileSharedLinkEffectivePermissionEnum `json:"effective_permission"`
	IsPasswordEnabled   bool                                  `json:"is_password_enabled"`
	Permissions         *FileSharedLinkPermissions            `json:"permissions,omitempty"`
	PreviewCount        int64                                 `json:"preview_count"`
	UnsharedAt          *time.Time                            `json:"unshared_at,omitempty"`
	URL                 string                                `json:"url"`
	VanityName          *string                               `json:"vanity_name,omitempty"`
	VanityURL           *string                               `json:"vanity_url,omitempty"`
}

type FileTypeEnum string

const (
	FileTypeEnumFile FileTypeEnum = "file"
)

// File
// The bare basic representation of a file, the minimal
// amount of fields returned when using the `fields` query
// parameter.
type File struct {
	ContentCreatedAt  *time.Time             `json:"content_created_at,omitempty"`
	ContentModifiedAt *time.Time             `json:"content_modified_at,omitempty"`
	CreatedAt         time.Time              `json:"created_at"`
	CreatedBy         *UserMini              `json:"created_by,omitempty"`
	Description       string                 `json:"description"`
	Etag              *string                `json:"etag,omitempty"`
	FileVersion       *FileVersionMini       `json:"file_version,omitempty"`
	ID                string                 `json:"id"`
	ItemStatus        FileItemStatusEnum     `json:"item_status"`
	ModifiedAt        time.Time              `json:"modified_at"`
	ModifiedBy        FileUserBase           `json:"modified_by"`
	Name              *string                `json:"name,omitempty"`
	OwnedBy           FileUserBase           `json:"owned_by"`
	Parent            *FileFolderBase        `json:"parent,omitempty"`
	PathCollection    FilePathCollection     `json:"path_collection"`
	PurgedAt          *time.Time             `json:"purged_at,omitempty"`
	SequenceID        map[string]interface{} `json:"sequence_id"`
	Sha1              string                 `json:"sha1"`
	SharedLink        *FileSharedLink        `json:"shared_link,omitempty"`
	Size              int64                  `json:"size"`
	TrashedAt         *time.Time             `json:"trashed_at,omitempty"`
	Type              FileTypeEnum           `json:"type"`
}
